tests/test_visitors/test_ast/test_classes/test_getter_setter.py

```diff
@@ -0,0 +1,381 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.oop import (
+    UnpythonicGetterSetterViolation,
+)
+from wemake_python_styleguide.visitors.ast.classes import WrongClassVisitor
+
+module_getter_and_setter = """
+attribute = 1
+
+def get_attribute():
+    ...
+
+def set_attribute():
+    ...
+"""
+
+static_getter_and_setter = """
+attribute = 1
+
+class Test(object):
+    @staticmethod
+    def get_attribute():
+        ...
+
+    @staticmethod
+    def set_attribute():
+        ...
+"""
+
+property_getter_and_setter = """
+class Test(object):
+    def __init__(self):
+        self.attribute = 1
+
+    @property
+    def attribute(self):
+        ...
+
+    @property.setter
+    def attribute(self):
+        ...
+"""
+
+dataclass_property_getter_setter = """
+@dataclass
+class DataClass(object):
+    attribute: int
+
+    @property
+    def attribute(self):
+        ...
+
+    @property
```

Comment:
name `dataclass_property_getter_setter` suggest that it should be:

```suggestion
    @attribute.setter
```